app/models/product.rb

```diff
@@ -4,7 +4,22 @@ class Product < ApplicationRecord
   validates :price, presence: true
   validates :quantity, presence: true
 
-  belongs_to :category
-  belongs_to :user
+  scope :published, ->{ where(status: "published") }
+  scope :unpublished, ->{ where(status: "unpublished") }
+  scope :file, ->{ where(status: "file") }
+
+  enum status: {unpublished: 0, published: 1, file: 2} do
+    event :publish do
+      transition :unpublished => :published
+    end
 
+    event :archived do
+      transition :published => :file
```

Comment:
If I understood this right, for grammar correctness, this should be something like:

```
    event :archive do
      transition :published => :archived
```

In that sense, events are verbs, and transition states are adjectives

Reply:
And therefore it would also affect `scope :file` (it'd become `scope :archived`)